Blog: Using documentation for usability studies (#2685)
opensearch-project/project-website: _posts/2024-04-03-UsingDocumentationForUsabilityStudies.md

### What do you need to start a usability study?

There are some convenient tooling such as UserZoom (now UserTesting) or Lookback and others. These tools make it convenient to capture user screens, record their voice and facial expressions and integrate surveys after tasks. They also offer a variety of testing methodologies to incorporate in the design testing phase of product development. That said, a usability study can be conducted manually over any video conferencing software (such as Zoom). If no specialized tooling is used, make sure to record all calls, ask the user to screen share and to ask probing questions as relevant. If you automate your study to be run with just the prompts the software provides and you do not need to be present, it is called an unmoderated study. If on the other hand, you spend the time to walk through the experience and help with the questions and probing, it is called a moderated study. Moderated studies are more time consuming at the data collection stage, but can be an effective method for rapid prototyping if the team plans to incrementally and iteratively improve on the design. 

### Sample considerations

> When you run your study, sample considerations are important. Use you persona definitions to guide sample specifications for recruitment. For larger samples and basic usability concerns, consider obtaining a paid sample. Use your screener to qualify or disqualify participants for your study. If your experience is for a niche set of users, it will make sense to recruit them by leveraging your B2B sales and services partners. For open source projects consider blogging, dropping links on GitHub, your community forum, announcing on community meetings etc. Always validate your sample with the screening questions either to 

Reviewer: ```suggestion
When you run your study, sample considerations are important. Use your persona definitions to guide sample specifications for recruitment. For larger samples and basic usability concerns, consider obtaining a paid sample. Use your screener to qualify or disqualify participants for your study. If your experience is for a niche set of users, it will make sense to recruit them by leveraging your B2B sales and services partners.
```

> *On the “Questionnaire” page you can organize the questions for screening the participants, and also all other questions that accompany the tasks in your study:*

Reviewer: ```suggestion
On the **Questionnaire** page, shown in the following image, you can organize both the screening and task questions for your study.
```

![](/assets/media/blog-images/2024-04-03-UsingDocumentationForUsabilityStudies/Screenshot 2024-03-26 at 10.44.52 AM.png)
### Reporting results

> Once you have taken time to conduct the studies and obtain usage data, it is time to analyze and report results. The most important aspect to report is task pass or fails. A color coded table summarizing this is very informative. See an example below. In the table, the participant qualification as expert or proficient, other tooling proficiency and the result is called out to give the reader a context on the overall findings. 

Reviewer: ```suggestion
Once you have conducted your study and obtained usage data, you are ready to analyze that data and report the results. The most important data to report is whether or not the user successfully accomplished a given task. A color-coded table summarizing this information, shown in the following image, can be very helpful. 
```

![](/assets/media/blog-images/2024-04-03-UsingDocumentationForUsabilityStudies/image (3).png)

> In addition to tables, a good written report will include a write up of details on repetitive usage patterns, anomalies in user flows (i.e. the participant clicking around in areas that may not correct), or interpretations of icons and other design patterns. Embedding the report with user videos on points of failure or the user struggling to accomplish the task humanizes user experience research to your audience. 

Reviewer: ```suggestion
In addition to tables, a good written report will include details about repetitive usage patterns, anomalies in user flows, or interpretations of icons and other design patterns. Embedding the report with videos showing points of failure or the user struggling to accomplish a task can humanize user experience research for your audience. 
```

> You can also Figma to visually depict the effectiveness and time on task across participants or fragments of the usability study report as needed:

Reviewer: ```suggestion
You can also use Figma to visually depict participant task completion and time on task, as shown in the following image.
```

![](/assets/media/blog-images/2024-04-03-UsingDocumentationForUsabilityStudies/Screenshot 2024-03-26 at 10.45.21 AM.png)![](/assets/media/blog-images/2024-04-03-UsingDocumentationForUsabilityStudies/Screenshot 2024-03-26 at 10.45.57 AM.png)

### Automation of a usability study

> If you have predictable user flows that you are improving the UX on etc., it may make sense to automate the study. Consider using the same usability study set up, and the same required tasks. The only thing that you would update is your end point on the back end or the link the clickable prototype. 

Reviewer: ```suggestion
If you have predictable user flows for which you are improving the UX, it may make sense to automate the study. Consider using the same study setup and the same required tasks. Only your backend endpoint or the link to the clickable prototype would need to be changed. 
```

> ### Usability Study Toolkit

Reviewer: ```suggestion
### Usability study toolkit
```

> The illustrations above are offered to the open source community as a [usability study toolkit](https://www.figma.com/community/file/1354537053014589491). The toolkit includes essential set of components to plan and run the study including a questionnaire template and a task board to track the responses from participants. Organizing the assets in Figma will simplify the process of presenting your study to the stakeholders and reporting. The Usability study toolkit is available on [OpenSearch page on Figma community](https://www.figma.com/@OpenSearch), and it is available to all Figma users to share and adapt in their work for any purpose. 

Reviewer: ```suggestion
The preceding examples are offered to the open-source community as a [usability study toolkit](https://www.figma.com/community/file/1354537053014589491). The toolkit includes an essential set of components for planning and running a usability study, including a questionnaire template and a task board for tracking participant responses. Organizing your assets in Figma can simplify the process of presenting your study to stakeholders. The usability study toolkit is available on the [OpenSearch Project community Figma page](https://www.figma.com/@OpenSearch). 
```

![](/assets/media/blog-images/2024-04-03-UsingDocumentationForUsabilityStudies/Screenshot 2024-03-26 at 10.40.43 AM.png)

> *How to use the Figma file (Instructions):*

Reviewer: ```suggestion
The following image shows instructions on how to use the toolkit.
```

![](/assets/media/blog-images/2024-04-03-UsingDocumentationForUsabilityStudies/Screenshot 2024-03-26 at 10.41.39 AM.png)

> This video walks you through the [usability study toolkit](https://www.figma.com/community/file/1354537053014589491) that we have created for open source consumption.

Reviewer: ```suggestion
Additionally, the following video walks you through the toolkit.
```
